### PR TITLE
add extension methods for SetAppTheme and SetAppThemeColor

### PR DESCRIPTION
**Pull request by fredericoregateiro**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #
#117 
 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui.Markup/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/157 

 ### Additional information ###

### CODE REVIEW

**Comment by brminnick** [comment]
Thanks @fredericoregateiro! 

Your Proposal Champion is @bijington and he will help guide you through the [New Feature Proposal Process](https://github.com/CommunityToolkit/Maui.Markup#submitting-a-new-feature).

The Proposal for this feature, #117, must first be approved by the core maintainer team before we can review, approve and merge this PR. Shaun will bring this feature to a vote at our upcoming Monthly Standup meeting this Thursday.

After it is approved, we'll also need you to write the Docs by opening a PR on our Docs repo: https://github.com/MicrosoftDocs/CommunityToolkit/

I did take a quick look at this PR and noticed a few things missing:
- [ ] Add Unit Tests
- [ ] Update the Sample App to display the use of the new feature.

Thanks again for your hard work!

**Comment by fredericoregateiro** [comment]
Thanks @brminnick 
I'll be updating those taks as soon as i can

**Comment by bijington** [comment]
> Thanks @brminnick
> I'll be updating those taks as soon as i can

Thanks @fredericoregateiro, do let me know if you need any assistance with the tasks as I am here to assist in any way 👍

**Comment by jfversluis** [comment]
Docs are ready to go afaic

File: src/CommunityToolkit.Maui.Markup/Style.cs [hunk 1]
```diff
@@ -86,6 +86,34 @@ public Style<T> Add(params (BindableProperty Property, object Value)[] setters)
...
+		MauiStyle.Setters.Add(property, new AppThemeBinding { Light = light, Dark = dark });
+		return this;
+	}
+
+	/// <summary>
+	/// Adds the supplied <paramref name="setters"/> in an <see cref="AppThemeBinding"/>.
+	/// </summary>
+	/// <param name="setters">A set of <see cref="BindableProperty"/>, and value for light and dark theme.</param>
+	/// <returns>Style with added setters</returns>
+	public Style<T> Add(params (BindableProperty Property, object Light, object Dark)[] setters)
```
Comment (brminnick) [inline src/CommunityToolkit.Maui.Markup/Style.cs:107]: What do you think about changing the name of this method to something like `AddAppThemeBinding`?

My concern is that the method's API Definition is pretty vague, `Add(BindableProperty, object, object)`, and a dev may not read the variable names (`object light` and `object dark`) and subsequently miss use it.

File: src/CommunityToolkit.Maui.Markup/Style.cs [hunk 2]
```diff
@@ -86,6 +86,34 @@ public Style<T> Add(params (BindableProperty Property, object Value)[] setters)
...
 	}
 
+	/// <summary>
+	/// Adds the supplied <paramref name="light"/> and <paramref name="dark"/> values in an <see cref="AppThemeBinding"/>.
+	/// </summary>
+	/// <param name="property">"The <see cref="BindableProperty"/> to style </param>
+	/// <param name="light">"The light value for the <see cref="BindableProperty"/> </param>
+	/// <param name="dark">"The dark value for the <see cref="BindableProperty"/> </param>
+	/// <returns>Style with added setters</returns>
+	public Style<T> Add(BindableProperty property, object light, object dark)
```
Comment (brminnick) [inline src/CommunityToolkit.Maui.Markup/Style.cs:96]: What do you think about changing the name of this method to something like `AddAppThemeBinding`?

My concern is that the method's API Definition is pretty vague, `Add(BindableProperty, object, object)`, and a dev may not read the variable names (`object light` and `object dark`) and subsequently miss use it.

**Review by brminnick — APPROVED** [review]
Approved!

Thanks so much @fredericoregateiro + @bijington!!!

**Pull request merged**